src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -3578,6 +3578,22 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
             if (ASRUtils::is_integer(*t) ||
                 ASR::is_a<ASR::Logical_t>(*ASRUtils::type_get_past_pointer(t))) {
                 switch( a_kind ) {
+                    case 1 : {
+                        if (llvm::ConstantInt* CI = llvm::dyn_cast<llvm::ConstantInt>(tmp)) {
+                            if (CI->getBitWidth() <= 8) {
+                                if (CI->isZero())
```

Comment:
The other issue is that this assumes a constant (compile time?) integer. I think this will not work for printing a variable. One has to do this "if" at runtime.

Reply:
+1

Reply:
@certik How do I access the print function in runtime? The only reference I see to a print function is the `lfortran_printf` function.

Reply:
We either have to create a runtime function that prints this correctly, or we have to insert this logic using LLVM commands. Writing a function seems easier. There might even be a way to make `printf` print True/False based on an integer value 1,0.

Reply:
Can you guide me regarding how and where to create the function. I'm guessing inside the `runtime` folder?

Reply:
Do `git grep lfortran_printf`, find the function, then add another one that creates the proper True/False based on an int, and then call it from llvm to create True/False at runtime.

Reply:
What about creating an array ([ArrayType](https://llvm.org/doxygen/classllvm_1_1ArrayType.html)) of `i8*` type in LLVM, `{"False", "True"}`. Then use, `CreateGEP` along with `tmp` to access 0-index or 1-index from the above value at run time.

Reply:
That should also work.

Reply:
@kinshukdua -- are you willing to complete this? Or, may I take over?

Reply:
@kinshukdua -- are you willing to complete this? Or, may I take over?

Reply:
I am not experienced either.  Let's try https://github.com/lcompilers/lpython/pull/357#discussion_r848264865 ?